src/components/Form.js

```diff
@@ -3,77 +3,153 @@ import TextInput from './TextInput';
 import Datepicker from './Datepicker';
+import Button from './Button';
 
-const Form = props => {
-  let propsArray = [
-    {
-      title: 'Event Name',
-      name: 'eventName',
-      inputType: 'text'
-    },
-    {
-      title: 'Event Website',
-      name: 'eventWebsite',
-      inputType: 'text'
-    },
-    {
-      title: 'Event Date',
-      name: 'eventDate'
-    },
-    {
-      title: 'Event Location'
-    },
-    {
-      title: 'Submission Due Date',
-      name: 'submissionDueDate'
-    },
-    {
-      title: 'Submission Website',
-      name: 'submissionWebsite'
-    },
-    {
-      name: 'isCompensated',
-      question: 'Are all speakers compensated at your event?',
-      options: ['Yes', 'No']
-    },
-    {
-      name: 'hasCodeOfConduct',
-      question: 'Does your event have a publicly visible code of conduct?',
-      options: ['Yes', 'No']
-    },
-    {
-      name: 'hasDiversityScholarships',
-      question: 'Does your event provide diversity scholarships?',
-      options: ['Yes', 'No']
-    }
-  ];
-  const handleChange = e => {
-    console.log(e.target.name, e.target.value);
-  };
-
-  const handleAutocomplete = city => {
-    console.log(city);
-  };
-  const handleDatepicker = selectedDate => {
-    console.log(selectedDate);
-  };
-  propsArray = propsArray.map(current => ({ ...current, handleChange }));
-  propsArray[2] = { ...propsArray[2], handleDatepicker };
-  propsArray[3] = { ...propsArray[3], handleAutocomplete };
-  propsArray[4] = { ...propsArray[4], handleDatepicker };
-  return (
-    <form>
-      <TextInput {...propsArray[0]} />
-      <TextInput {...propsArray[1]} />
-      <Datepicker {...propsArray[2]} />
-      <Autocomplete {...propsArray[3]} />
-      <Datepicker {...propsArray[4]} />
-      <TextInput {...propsArray[5]} />
-      <RadioButton {...propsArray[6]} />
-      <RadioButton {...propsArray[7]} />
-      <RadioButton {...propsArray[8]} />
-    </form>
-  );
-};
+class Form extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      eventName: '',
+      eventWebsite: '',
+      eventDate: '',
+      eventLocation: '',
+      submissionDueDate: '',
+      submissionWebsite: '',
+      isCompensated: '',
+      hasCodeOfConduct: '',
+      hasDiversityScholarships: '',
+      contactName: '',
+      contactEmail: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleAutocomplete = this.handleAutocomplete.bind(this);
+    this.handleDatepicker = this.handleDatepicker.bind(this);
+    this.showState = this.showState.bind(this);
+  }
+  showState() {
+    alert(this.state);
+  }
+  handleChange(e) {
+    this.setState({ [e.target.name]: e.target.value });
+  }
 
+  handleAutocomplete(city) {
+    this.setState({ eventLocation: city });
+  }
+  handleDatepicker(selectedDate, dateName) {
+    this.setState({ [dateName]: selectedDate });
+  }
+  makePropsArray() {
+    let propsArray = [
```

Comment:
Since this function is only creating an array, it doesn't necessarily need to be inside of a separate function. Asked the bridge mentors, and they recommended that we store this in a separate json file (formConfig.json for example), and pass it in as a prop to the `Form` component.

It can be turned into an object with keys that match the field that the props are for. This will help to make it more readable when spreading the props to their corresponding inputs. 

Example:
```js
const propsArray = {
  eventName: {
    title: 'Event Name',
    name: 'eventName',
    inputType: 'text'
  },
  eventWebsite: {
    title: 'Event Website',
    name: 'eventWebsite',
    inputType: 'text'
  }
  /* ...and the rest of them */
}